src/components/GitHubAuth.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import GitHubLogin from 'react-github-login';
+import { postGithubCode } from '../actions/login'
+import { withRouter } from 'react-router';
+import { Icon, Message } from 'semantic-ui-react'
+
+
+class GitHubAuth extends Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            code: '',
+            error: null,
+            posted: false,
+            errorText: ''
+        }
+        this.postCode = this.postCode.bind(this);
+        this.GithubError = this.GithubError.bind(this);
+    }
+
+    postCode = (response) => {
+        console.log(response.code)
+        this.setState({
+            code: response.code
+        })
+        const data = {
+            code: this.state.code
+        }
+        this.props.postGithubCode(data, this.callback)
+        this.setState({
+            code: ''
+        })
+    }
+
+    callback = () => {
+        this.setState({
+            error: this.props.loginerror ? true : false,
+            posted: true,
+            errorText: 'Server Error: try again.'
+        })
+        console.log(this.state.error, this.state.posted);
+        if (!this.state.error) {
+            this.props.history.push('/')
+        }
+        setTimeout(() => {
+            this.setState({
+                error: false,
+                posted: false,
+            })
+        }, 4000)
+    }
+
+    icon = () => {
```

Comment:
Given that there is another PR for Google Oauth I think it's logical to rename icon as githubIcon. 
```suggestion
    githubIcon = () => {
```